browser/brave_shields/https_upgrade_browsertest.cc

```diff
@@ -0,0 +1,181 @@
+        ->SetIsReadyForTesting();
+    // By default allow all hosts on HTTPS.
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
+    host_resolver()->AddRule("*", "127.0.0.1");
+
+    // Set up "insecure.test" as a hostname with an SSL error. kHttps upgrades
+    // to this host will fail, (or fall back in some cases).
+    scoped_refptr<net::X509Certificate> cert(https_server_.GetCertificate());
+    net::CertVerifyResult verify_result;
+    verify_result.is_issued_by_known_root = false;
+    verify_result.verified_cert = cert;
+    verify_result.cert_status = net::CERT_STATUS_COMMON_NAME_INVALID;
+    for (const std::string& host :
+         {"insecure1.test", "insecure2.test", "insecure3.test"}) {
+      mock_cert_verifier_.mock_cert_verifier()->AddResultForCertAndHost(
+          cert, host, verify_result, net::ERR_CERT_INVALID);
+    }
+
+    http_server_.AddDefaultHandlers(GetTestDataDir());
+    https_server_.AddDefaultHandlers(GetTestDataDir());
+    ASSERT_TRUE(http_server_.Start());
+    ASSERT_TRUE(https_server_.Start());
+
+    HttpsOnlyModeUpgradeInterceptor::SetHttpsPortForTesting(
+        https_server()->port());
+    HttpsOnlyModeUpgradeInterceptor::SetHttpPortForTesting(
+        http_server()->port());
+  }
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    PlatformBrowserTest::SetUpCommandLine(command_line);
+    mock_cert_verifier_.SetUpCommandLine(command_line);
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    PlatformBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
+  }
+
+  void AttemptToNavigateToURL(const GURL& url) {
+    content::NavigateToURLBlockUntilNavigationsComplete(Contents(), url, 1,
+                                                        true);
+  }
+
+  content::WebContents* Contents() {
+    return chrome_test_utils::GetActiveWebContents(this);
+  }
+
+  HostContentSettingsMap* ContentSettings() {
+    return HostContentSettingsMapFactory::GetForProfile(
+        chrome_test_utils::GetProfile(this));
+  }
+
+ protected:
+  net::EmbeddedTestServer* http_server() { return &http_server_; }
+  net::EmbeddedTestServer* https_server() { return &https_server_; }
+
+ private:
+  base::test::ScopedFeatureList feature_list_;
+  net::EmbeddedTestServer http_server_{net::EmbeddedTestServer::TYPE_HTTP};
+  net::EmbeddedTestServer https_server_{net::EmbeddedTestServer::TYPE_HTTPS};
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// If the user navigates to an HTTP URL for a site that supports HTTPS, the
+// navigation should end up on the HTTPS version of the URL.
+IN_PROC_BROWSER_TEST_F(HttpsUpgradeBrowserTest, CheckUpgrades) {
```

Comment:
I'd suggest copy/pasting this test for two other cases that we should ensure never break:
1. The feature is disabled: nothing gets upgraded to HTTPS
2. Shields is disabled for each site: none of the sites get upgraded to HTTPS

Reply:
Added 1 and 2.